### PR TITLE
Update Spotify to work with new API.

### LINKED ISSUE #400: Spotify integration broken (possible fix concept included)
I'm pretty sure the reason why it no longer works is cause Spotify removed its local API in favour of its web based one. 

I found this out cause Snip (app that fetches Spotify playing info) no longer works, but someone made a c# fix for it on the Snip repo https://github.com/dlrudie/Snip/pull/303

So, the solution is to find/create a java implementation of said API and use it to get info instead of local API.

### PR DESCRIPTION
**Pull request by ConorTheDev**
## Description  
Fixed Spotify so it works with the new API

## Context  
Fixes #400 by using access tokens which you can get from using the little website I created at:
https://spotifymod-client.herokuapp.com

## Checklist  
- ✅ All *new* Java and Kotlin files have the right copyright header  
- ✅ I have tested my code by building it locally  
- ✅ This is not a work-in-progress and is ready for merge  
- ✅ I am submitting the PR under and understand the terms of the GNU Lesser General Public License v3.0

### CODE REVIEW

**Comment by RDIL** [comment]
Finally a fix!  Thanks Conor :D

**Comment by ConorTheDev** [comment]
> Finally a fix! Thanks Conor :D

Been requested for a while, thought I'd fix it as the Mod version wasn't going to work for issues related to forge. Decided to put it in Hyperium.

**Comment by asbyth** [comment]
spotify is back :crab: :crab: :crab: :crab:

**Comment by RDIL** [comment]
Yeet

File: src/main/java/cc/hyperium/mods/spotify/SpotifyGui.java [hunk 1]
```diff
@@ -107,6 +138,12 @@ protected void actionPerformed(GuiButton button) {
             case 1:
                 this.controls.setX(0);
                 this.controls.setY(0);
+                break;
+            case 2:
+                try {
+                    Desktop.getDesktop().browse(new URI("http://spotifymod-client.herokuapp.com"));
```
Comment (RDIL) [inline src/main/java/cc/hyperium/mods/spotify/SpotifyGui.java]: reeeee no https? :(

File: src/main/java/cc/hyperium/config/Settings.java [hunk 2]
```diff
@@ -403,10 +390,12 @@
     public static String PAUSE_STYLE = GuiStyle.HYPERIUM.name();
 
     @ConfigOpt
+    @ToggleSetting(name = "gui.spotify.spotifynotifications", category = SPOTIFY)
     public static boolean SPOTIFY_NOTIFICATIONS = false;
 
     @ConfigOpt
-    public static boolean SPOTIFY_FORCE_DISABLE = true;
+    @ToggleSetting(name = "gui.spotify.forcedisable", category = SPOTIFY)
+    public static boolean SPOTIFY_FORCE_DISABLE = false;
```
Comment (RDIL) [inline src/main/java/cc/hyperium/config/Settings.java]: I don't think the force disable option should exist anymore since spotify works now.

Comment (asbyth) [reply src/main/java/cc/hyperium/config/Settings.java]: existed before it broke, why not keep it

File: src/main/java/cc/hyperium/mods/spotify/SpotifyGui.java (hunk 1 above)
Comment (ConorTheDev) [reply src/main/java/cc/hyperium/mods/spotify/SpotifyGui.java]: I'll fix that now

Comment (Cubxity) [reply src/main/java/cc/hyperium/mods/spotify/SpotifyGui.java]: @ConorTheDev is this resolved?

**Comment by RoccoDev** [comment]
That's not a problem if you ping it regularly with a service (e.g, Uptime Robot).
However, keep in mind free accounts only get 550 hours per month (if you add a credit card you get 450 extra hours).

**Comment by asbyth** [comment]
closing since mediamod :)

**Pull request closed without merging**